supervisor/utils/systemd_journal.py

```diff
@@ -0,0 +1,114 @@
+"""Utilities for working with systemd journal export format."""
+from collections.abc import AsyncGenerator
+from datetime import UTC, datetime
+from functools import wraps
+
+from aiohttp import ClientResponse
+
+from supervisor.host.const import LogFormatter
+
+
+class MalformedBinaryEntry(Exception):
+    """Raised when binary entry in the journal isn't followed by a newline."""
```

Comment:
Sure. Shouldn't it also be a subclass of HassioError then, so it's handled more gracefully in the API?

Reply:
Oh yes definitely. I didn't notice that. Please make it a subclass of `HassioError` otherwise we'll treat it as an unhandled internal error everywhere.

Reply:
Done, I've also adjusted the name and added a test for it.